apps/babelstream/package.py

```diff
@@ -0,0 +1,268 @@
+        if "+sycl" in spec:
+            # args.append("-DCXX_EXTRA_FLAGS= -fsycl -O3")
+            #this is required to enable -DCMAKE_CXX_COMPILER=icpx flag from CMake
+            args.append("-DSYCL_COMPILER=" + spec.variants['compiler'].value)
+
+
+        #===================================
+        #             SYCL 2020 
+        #===================================
+        
+        #cmake -Bbuild -H. -DMODEL=sycl2020 -DSYCL_COMPILER=ONEAPI-ICPX -DSYCL_COMPILER_DIR=ONEAPI-ICPX
+        if "+sycl2020" in spec:
+            # -fsycl flag is required for setting up sycl/sycl.hpp seems like it doesn't get it from the CMake file
+            args.append("-DCXX_EXTRA_FLAGS= -fsycl -O3")
+            #this is required to enable -DCMAKE_CXX_COMPILER=icpx flag from CMake
+            args.append("-DSYCL_COMPILER=ONEAPI-ICPX")
+
+        #===================================
+        #             HIP(ROCM) 
+        #===================================
+        
+        if "+rocm" in spec:
+            hip_comp = self.spec["rocm"].prefix + "/bin/hipcc"
+            args.append("-DCMAKE_CXX_COMPILER=" + hip_comp)
+            
+            rocm_archs = spec.variants['amdgpu_target'].value
+            if 'none' not in rocm_archs:
+                args.append(
+                    "-DCXX_EXTRA_FLAGS= --offload-arch="
+                    + self.spec.variants["amdgpu_target"].value
+                    + " "
+                    + self.spec.variants["flags"].value
+                    + " -O3"
+                )
+
+        #===================================
+        #             TBB 
+        #===================================
+
+        if "+tbb" in spec:
+            args.append("-DTBB_ROOT=" + self.spec["tbb"].prefix + "/lib/cmake/tbb")
+            # Hack to get around the fact that TBBConfig.cmake is linking 32bit libs.
+            args.append("-DCMAKE_SIZEOF_VOID_P=8")
+
+        #===================================
+        #             OpenCL (ocl) 
+        #===================================
+        if "+ocl" in spec:
+            args.append("-DCMAKE_CXX_COMPILER=" + self.compiler.cxx)
+            ocl_lib = self.spec["pocl"].prefix + "/lib64/libOpenCL.so"
+            args.append("-DOpenCL_LIBRARY=" + ocl_lib)
+            
+
+
+        #===================================
+        #             RAJA 
+        #===================================
+        if "+raja" in spec:
+            args.append("-DCMAKE_CXX_COMPILER=" + self.compiler.cxx)
+            args.append("-DRAJA_IN_TREE=" + spec.variants['dir'].value)
+            # args.append("-DRAJA_DIR=" + self.spec["raja"].prefix + "/share ")
+            if "+omp" in spec:
+                args.append("-DENABLE_OPENMP=ON")
+            if "+cuda" in spec:
+                args.append("-DENABLE_OPENMP=ON")
+    
+    
+        #===================================
+        #             kokkos 
+        #===================================
+
+        if "+kokkos" in spec:
```

Comment:
Can you add a comment saying "remember to set the arch via the usual kokkos method e.g., ..."?

Reply:
Kokkos for nvidia actually needs more flags to be set because it complains about arch flags need to be set for instance `Kokkos_ARCH_VOLTA72`  so I will code their conditions and flags here to